### ISSUE TITLE
Building for illumos fails with the latest hybrid cache change

### ISSUE THREAD

**Issue by zllovesuki** [Yiling-J/theine-go#27]
```
CGO_ENABLED=0 GOOS=illumos GOARCH=amd64 GOARM=7 GOAMD64=v3 go build -tags 'osusergo netgo urfave_cli_no_docs no_mocks' -ldflags "-s -w -extldflags -static -X kon.nect.sh/specter/cmd/specter.Build=d768c18 -X kon.nect.sh/specter/acme.BuildTime=1687420223" -o bin/specter-illumos-amd64 .
# github.com/ncw/directio
../../go/pkg/mod/github.com/ncw/directio@v1.0.5/direct_io_unix.go:22:35: undefined: syscall.O_DIRECT
```

**Comment by Yiling-J** [comment]
Thanks for reporting! I think that's because O_DIRECT flag is not supported on illumos(or Go doesn't include that flag yet), so I need to take a look and maybe patch the directio package I use. Before I fix it you can use v0.2.9, 0.3.0/0.3.1 are all related to hybrid cache.

**Comment by Yiling-J** [comment]
@zllovesuki Could you please switch to the main branch and try building again?
